### ISSUE TITLE
[BUG] Issues adding 2nd and 3rd node to harvester test cluster

### ISSUE THREAD

**Issue by poldiderjungdrache** [harvester/harvester#2169]
**Describe the bug**

I am trying to create a small harvester cluster. I installed the master node without issues. Now I am installing additional 2nd and 3rd node. The install using 1.0.1 ISO shows no issues, however both nodes remain in status NotReady

**To Reproduce**
1. install first node using ISO
2. Install 2nd and 3rd node, selecting Join existing cluster in installer
3. provide cluster token, VIP etc. as requested by the installer
4. node remains in status NotReady and never joins the cluster

**Expected behavior**

Nodes should join cluster

**Support bundle**

[supportbundle_17ecbbe0-3abd-43f0-9267-4cbb33e6282d_2022-04-19T20-08-52Z.zip](https://github.com/harvester/harvester/files/8516191/supportbundle_17ecbbe0-3abd-43f0-9267-4cbb33e6282d_2022-04-19T20-08-52Z.zip)

**Environment:**
 - Harvester ISO version: 1.0.1
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): baremetal (2 x HP ProDesk 400 G6) + 1 virtual

**Additional context**
Troubleshooting done:

Bootstrap output: 
-- Logs begin at Tue 2022-04-19 19:14:40 UTC, end at Tue 2022-04-19 21:14:21 UTC. --
Apr 19 21:14:15 harvester03 systemd[1]: Starting Rancher Bootstrap...
Apr 19 21:14:15 harvester03 rancherd[1737]: time="2022-04-19T21:14:15Z" level=info msg="Bootstrapping Rancher (master-head/v1.21.11+rke2r1)"
Apr 19 21:14:15 harvester03 rancherd[1737]: time="2022-04-19T21:14:15Z" level=info msg="Writing plan file to /var/lib/rancher/rancherd/plan/plan.json"
Apr 19 21:14:15 harvester03 rancherd[1737]: time="2022-04-19T21:14:15Z" level=info msg="Applying plan with checksum "
Apr 19 21:14:15 harvester03 rancherd[1737]: time="2022-04-19T21:14:15Z" level=info msg="No image provided, creating empty working directory /var/lib/rancher/rancherd/plan/work/20220419-211415-applied.plan/_0"
Apr 19 21:14:15 harvester03 rancherd[1737]: time="2022-04-19T21:14:15Z" level=info msg="Running command: /usr/bin/env [sh /var/lib/rancher/rancherd/install.sh]"
Apr 19 21:14:15 harvester03 rancherd[1737]: time="2022-04-19T21:14:15Z" level=info msg="[stdout]: [INFO]  Using default agent configuration directory /etc/rancher/agent"
Apr 19 21:14:15 harvester03 rancherd[1737]: time="2022-04-19T21:14:15Z" level=info msg="[stdout]: [INFO]  Using default agent var directory /var/lib/rancher/agent"
Apr 19 21:14:15 harvester03 rancherd[1737]: time="2022-04-19T21:14:15Z" level=info msg="[stderr]: [WARN]  /usr/local is read-only or a mount point; installing to /opt/rancher-system-agent"
Apr 19 21:14:16 harvester03 rancherd[1737]: time="2022-04-19T21:14:16Z" level=info msg="[stdout]: [INFO]  Determined CA is necessary to connect to Rancher"
Apr 19 21:14:16 harvester03 rancherd[1737]: time="2022-04-19T21:14:16Z" level=info msg="[stdout]: [INFO]  Successfully downloaded CA certificate"
Apr 19 21:14:16 harvester03 rancherd[1737]: time="2022-04-19T21:14:16Z" level=info msg="[stdout]: [INFO]  Value from https://harvester.tianmimi.infra:443/cacerts is an x509 certificate"
Apr 19 21:14:16 harvester03 rancherd[1737]: time="2022-04-19T21:14:16Z" level=info msg="[stdout]: [INFO]  Successfully tested Rancher connection"
Apr 19 21:14:16 harvester03 rancherd[1737]: time="2022-04-19T21:14:16Z" level=info msg="[stdout]: [INFO]  Downloading rancher-system-agent from https://harvester.tianmimi.infra:443/assets/rancher-system-agent-amd64"
Apr 19 19:15:37 harvester03 rancherd[1737]: time="2022-04-19T19:15:37Z" level=info msg="[stdout]: [INFO]  Successfully downloaded the rancher-system-agent binary."
Apr 19 19:15:37 harvester03 rancherd[1737]: time="2022-04-19T19:15:37Z" level=info msg="[stdout]: [INFO]  Generating Cattle ID"
Apr 19 19:15:37 harvester03 rancherd[1737]: time="2022-04-19T19:15:37Z" level=info msg="[stdout]: [INFO]  Successfully downloaded Rancher connection information"
Apr 19 19:15:37 harvester03 rancherd[1737]: time="2022-04-19T19:15:37Z" level=info msg="[stdout]: [INFO]  systemd: Creating service file"
Apr 19 19:15:37 harvester03 rancherd[1737]: time="2022-04-19T19:15:37Z" level=info msg="[stdout]: [INFO]  Creating environment file /etc/systemd/system/rancher-system-agent.env"
Apr 19 19:15:37 harvester03 rancherd[1737]: time="2022-04-19T19:15:37Z" level=info msg="[stdout]: [INFO]  Enabling rancher-system-agent.service"
Apr 19 19:15:37 harvester03 rancherd[1737]: time="2022-04-19T19:15:37Z" level=info msg="[stderr]: Created symlink /etc/systemd/system/multi-user.target.wants/rancher-system-agent.service → /etc/systemd/system/rancher-system-agent.service."
Apr 19 19:15:38 harvester03 rancherd[1737]: time="2022-04-19T19:15:38Z" level=info msg="[stdout]: [INFO]  Starting/restarting rancher-system-agent.service"
Apr 19 19:15:38 harvester03 systemd[1]: rancherd.service: Succeeded.
Apr 19 19:15:38 harvester03 systemd[1]: Finished Rancher Bootstrap.

console.log (just repeats the below message):

time="2022-04-19T20:14:18Z" level=error msg="exit status 1%!(EXTRA string=)"
time="2022-04-19T20:14:18Z" level=error msg="exit status 1The connection to the server 127.0.0.1:6443 was refused - did you specify the right host or port?\n"

**Comment by poldiderjungdrache** [comment]
Guess I found some issue:

rancher-system-agent is in a restart loop:

Apr 19 19:15:48 harvester03 systemd[1]: rancher-system-agent.service: Failed with result 'exit-code'.
Apr 19 19:15:53 harvester03 systemd[1]: rancher-system-agent.service: Scheduled restart job, restart counter is at 3.
Apr 19 19:15:53 harvester03 systemd[1]: Stopped Rancher System Agent.
Apr 19 19:15:53 harvester03 systemd[1]: Started Rancher System Agent.
Apr 19 19:15:53 harvester03 rancher-system-agent[2041]: time="2022-04-19T19:15:53Z" level=info msg="Rancher System Agent version v0.2.3 (00181cd) is starting"
Apr 19 19:15:53 harvester03 rancher-system-agent[2041]: time="2022-04-19T19:15:53Z" level=info msg="Using directory /var/lib/rancher/agent/work for work"
Apr 19 19:15:53 harvester03 rancher-system-agent[2041]: time="2022-04-19T19:15:53Z" level=info msg="Starting remote watch of plans"
Apr 19 19:15:54 harvester03 rancher-system-agent[2041]: time="2022-04-19T19:15:54Z" level=info msg="Initial connection to Kubernetes cluster failed with error Get \"https://harvester.tianmimi.infra/version\": x509: certificate signed by unknown authori>
Apr 19 19:15:54 harvester03 rancher-system-agent[2041]: panic: error while connecting to Kubernetes cluster with nullified CA data: Get "https://harvester.tianmimi.infra/version": x509: certificate signed by unknown authority

He doesn't trust my internal CA ... well of course.

I added the CA to opensuse using update-ca-certificates but that did not impress him. 
Is there a way to tell him to ignore certificate validation issues?
